packages/babel-core/test/config-loading.js

```diff
@@ -188,25 +188,26 @@ describe("@babel/core config loading", () => {
       const options2 = loadConfig(opts).options;
       expect(options2.plugins.length).toBe(options1.plugins.length);
 
+      expect(options2.plugins[0]).not.toBe(options1.plugins[0]);
       for (let i = 0; i < options1.plugins.length; i++) {
         if (i === 0) {
-          expect(options2.plugins[i]).not.toBe(options1.plugins[i]);
-        } else {
-          expect(options2.plugins[i]).toBe(options1.plugins[i]);
+          continue;
         }
+        expect(options2.plugins[i]).toBe(options1.plugins[i]);
       }
 
       process.env.INVALIDATE_PLUGIN3 = true;
 
       const options3 = loadConfig(opts).options;
       expect(options3.plugins.length).toBe(options1.plugins.length);
-
+      for (const i of [0, 5]) {
+        expect(options3.plugins[i]).not.toBe(options1.plugins[i]);
+      }
```

Comment:
Nit, but I find it clearer to be more explicit in tests when it's not too long:
```suggestion
      expect(options3.plugins[0]).not.toBe(options1.plugins[0]);
      expect(options3.plugins[5]).not.toBe(options1.plugins[5]);
```